tracpro/templates/polls/poll_read.html

```diff
@@ -114,37 +115,44 @@
     {% endif %}
 
   {% endfor %}
-{% endblock %}
-{% block extra-script %}
-<script type='text/javascript'>
-    // <![CDATA[
+
+<script>
+    $(function () {
+      // Run this once, in the case that the user has
+      // hit the Back button.
+      onValueTypeChange();
+    });
     function onWindowChange(ctrl) {
       ctrl.form.submit();
     }
-    function onValueTypeChange(ctrl) {
+    function onValueTypeChange() {
       valueType = $("#value-type-select :selected").text().toLowerCase();
+      changeChartValueType(valueType);
+      // Keep track of this value in a "cached" hidden variable for Back button click
+      $("#value-type-cache").val(valueType);
```

Comment:
I see that you are setting the value of the cache here, but I don't see where it is used.